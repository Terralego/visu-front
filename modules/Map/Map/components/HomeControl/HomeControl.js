import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '@blueprintjs/core';

import AbstractMapControl from '../../../helpers/AbstractMapControl';
import translateMock from '../../../../../utils/translate';

export class HomeControl extends AbstractMapControl {
  static containerClassName = 'mapboxgl-ctrl mapboxgl-ctrl-group mapboxgl-ctrl-home';

  static propTypes = {
    /** Function used to translate wording. Takes key and object of options as parameters */
    translate: PropTypes.func,
  }

  static defaultProps = {
    translate: translateMock({
      'terralego.map.home_control.button_label': 'Home',
    }),
  }

  goHome = () => {
    const { map, fitBounds, center, zoom = 7 } = this.props;
    if (fitBounds) {
      map.fitBounds(fitBounds, {
        padding: 10,
      });
      return;
    }

    if (center) {
      map.flyTo({ center, zoom });
    }
  }

  render () {
    const { translate } = this.props;

    return (
      <button
        className="mapboxgl-ctrl-icon"
        type="button"
        onClick={this.goHome}
        title={translate('terralego.map.home_control.button_label')}
        aria-label={translate('terralego.map.home_control.button_label')}
      >
        <Icon icon="home" />
      </button>
    );
  }
}

export default HomeControl;
