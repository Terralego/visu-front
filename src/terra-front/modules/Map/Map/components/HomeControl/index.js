import HomeControl from './HomeControl';

export default HomeControl;
