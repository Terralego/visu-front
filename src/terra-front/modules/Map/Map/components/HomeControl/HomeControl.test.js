import React from 'react';
import renderer from 'react-test-renderer';

import HomeControl from './HomeControl';

it('should render', () => {
  const tree = renderer.create(<HomeControl />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should go home with bounds', () => {
  const map = {
    fitBounds: jest.fn(),
  };

  const instance = new HomeControl({ map, fitBounds: [1, 2], fitBoundsParams: { padding: 10 } });
  instance.goHome();
  expect(map.fitBounds).toHaveBeenCalledWith([1, 2], { padding: 10 });
});

it('should go home with center', () => {
  const map = {
    flyTo: jest.fn(),
  };

  const instance = new HomeControl({ map, center: [1, 2] });
  instance.goHome();
  expect(map.flyTo).toHaveBeenCalledWith({ center: [1, 2], zoom: 7 });
});

it('should not go home', () => {
  const map = {
    flyTo: jest.fn(),
    fitBounds: jest.fn(),
  };

  const instance = new HomeControl({ map });
  instance.goHome();
  expect(map.flyTo).not.toHaveBeenCalled();
  expect(map.fitBounds).not.toHaveBeenCalled();
});
