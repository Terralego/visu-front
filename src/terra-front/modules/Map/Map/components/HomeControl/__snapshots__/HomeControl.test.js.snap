// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <button
      aria-label="Home"
      className="mapboxgl-ctrl-icon"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-home"
        icon="home"
      >
        <svg
          data-icon="home"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            home
          </desc>
          <path
            d="M2 10v5c0 .55.45 1 1 1h3v-5h4v5h3c.55 0 1-.45 1-1v-5L8 4l-6 6zm13.71-2.71L14 5.59V2c0-.55-.45-1-1-1s-1 .45-1 1v1.59L8.71.29C8.53.11 8.28 0 8 0s-.53.11-.71.29l-7 7a1.003 1.003 0 001.42 1.42L8 2.41l6.29 6.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </span>
</span>
`;
